WARNINGS.md

```diff
@@ -796,6 +797,38 @@ may never see the warning.
 
 --------------------------------------------------------------------------------
 
+## <a name="provider-params"></a>Calls to 'provider' should specify a list of fields and a documentation
+
+  * Category name: `provider-params`
+  * Automatic fix: no
+
+Calls to `provider` should specify a documentation string and a list of fields:
+
+```python
+fooInfo = provider(
+    doc = "Some documentation.",
+    fields = ["field1", "field2"],
+)
+```
+
+Fields should also be documented when needed:
+
+```python
+provider(
+    doc = "Some documentation.",
+    fields = {
+        "field1": "Documentation for field1",
+        "field2': "Documentation for field2",
+    })
+``
+
+Note that specifying a list of fields is a breaking change. It is an error if a
```

Comment:
Does an explicit fields=None work to (a) indicate a schemaless struct and (b) suppress the warning? If so, we should document it. If not, we should implement it (and document it).

Reply:
Yes, it's supported. I've updated the text here.

We could actually declare `struct` as `struct = provider()`.